.github/scripts/auto-backport.py

```diff
@@ -150,7 +150,10 @@ def main():
         start_commit, end_commit = args.commits.split('..')
         commits = repo.compare(start_commit, end_commit).commits
         for commit in commits:
-            for pr in commit.get_pulls():
+            match = re.search(rf"Closes .*#([0-9]+)", commit.commit.message, re.IGNORECASE)
+            if match:
+                pr_number = int(match.group(1))
+                pr = repo.get_pull(pr_number)
```

Comment:
Isn't this only the case for single commit PRs? Does this work for multi-commit PRs as well?
I don't remember if we use `Closes` there as well.

Reply:
No, it is working also for multi-commit PRS
See results when i was running it locally 
```
[yaronkaikov@london]~/git/scylladb (master)$ python .github/scripts/auto-backport.py --repo scylladb/scylladb --base-branch refs/heads/master --commits afd1fc8e9f149143a394dff30c7f13863a7856d3..6efde20939d072406b6e80a7ece1026134b0268a
no backport label: 21324
2024-11-06 22:48:08,182 - INFO - Found PR #21401 with commit ['77604b4ac76b31c4a6433219cb3bd9792a9bf434'] and the following labels: ['backport/6.1', 'backport/6.2']
2024-11-06 22:48:45,663 - WARNING - Cherry-pick conflict on commit 77604b4ac76b31c4a6433219cb3bd9792a9bf434: Cmd('git') failed due to: exit code(1)
  cmdline: git cherry-pick 77604b4ac76b31c4a6433219cb3bd9792a9bf434 -m1 -x
  stdout: 'CONFLICT (modify/delete): .github/scripts/auto-backport.py deleted in HEAD and modified in 77604b4ac7 (.github/script/auto-backport.py: push backport PR to `scylladbbot` fork).  Version 77604b4ac7 (.github/script/auto-backport.py: push backport PR to `scylladbbot` fork) of .github/scripts/auto-backport.py left in tree.'
  stderr: 'error: could not apply 77604b4ac7... .github/script/auto-backport.py: push backport PR to `scylladbbot` fork
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config advice.mergeConflict false"'
2024-11-06 22:48:48,748 - INFO - Pull request created: https://github.com/scylladb/scylladb/pull/21465
2024-11-06 22:48:49,658 - INFO - Assigned PR to original author: NamedUser(login="yaronkaikov")
2024-11-06 22:49:03,197 - WARNING - Cherry-pick conflict on commit 77604b4ac76b31c4a6433219cb3bd9792a9bf434: Cmd('git') failed due to: exit code(1)
  cmdline: git cherry-pick 77604b4ac76b31c4a6433219cb3bd9792a9bf434 -m1 -x
  stdout: 'CONFLICT (modify/delete): .github/scripts/auto-backport.py deleted in HEAD and modified in 77604b4ac7 (.github/script/auto-backport.py: push backport PR to `scylladbbot` fork).  Version 77604b4ac7 (.github/script/auto-backport.py: push backport PR to `scylladbbot` fork) of .github/scripts/auto-backport.py left in tree.'
  stderr: 'error: could not apply 77604b4ac7... .github/script/auto-backport.py: push backport PR to `scylladbbot` fork
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config advice.mergeConflict false"'
2024-11-06 22:49:06,177 - INFO - Pull request created: https://github.com/scylladb/scylladb/pull/21466
2024-11-06 22:49:07,161 - INFO - Assigned PR to original author: NamedUser(login="yaronkaikov")
no backport label: 21450
no backport label: 21440
```

PR 21450 for example is a multi-commit PR